FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -206,26 +206,30 @@ extension ComposeViewController {
         showSpinner("sending_title".localized)
         navigationItem.rightBarButtonItem?.isEnabled = false
 
-        composeMessageService.validateMessage(
+        let result = composeMessageService.validateMessage(
             input: input,
             contextToSend: contextToSend,
             email: email
         )
-        .publisher
-        .flatMap(encryptAndSend)
-        .receive(on: DispatchQueue.main)
-        .sinkFuture(
-            receiveValue: { [weak self] in
-                self?.handleSuccessfullySentMessage()
-            },
-            receiveError: { [weak self] error in
-                self?.handle(error: error)
-            })
-        .store(in: &cancellable)
-    }
-
-    private func encryptAndSend(_ message: SendableMsg) -> AnyPublisher<Void, ComposeMessageError> {
-        composeMessageService.encryptAndSend(message: message, threadId: input.threadId)
+        switch result {
+        case .success(let message):
+            encryptAndSend(message)
+        case .failure(let error):
+            handle(error: error)
+        }
+    }
+
+    private func encryptAndSend(_ message: SendableMsg) {
+        Task {
+            do {
+                try await composeMessageService.encryptAndSend(message: message, threadId: input.threadId)
+                handleSuccessfullySentMessage()
```

Comment:
Since every view controller is marked with `@MainActor` we could simply call methods from task and they will be invoked on the main thread.